pkg/mesh/backend.go

```diff
@@ -54,8 +56,9 @@ const (
 
 // Node represents a node in the network.
 type Node struct {
-	Endpoint     *wireguard.Endpoint
-	Key          []byte
+	KiloEndpoint *wireguard.Endpoint
```

Comment:
This new abstraction is pretty confusing to me. Why do nodes and peers have to have both a `KiloEndpoint` and an `Endpoint`? It seems like it will be very easy to forget to do some bookkeeping with these two pointers and break things. Semantically they are also very confusing because they are both "endpoints that Kilo uses", no? Maybe it would be easier to understand everything if we instead had:
```golang
type Node struct {
    DNS string
    Endpoint *net.UDPAddr
}
```

Reply:
Also, is the `Endpoint` field used outside of this package or can it be turned into a private field?

Reply:
If save DNS in a separate string, where would we save the port? Do you mean an addr field like `<dns name>:<port>`?